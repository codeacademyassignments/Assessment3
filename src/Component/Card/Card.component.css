.img {
  position: relative;
  width:  100%;
  height: 303px;
  background-position: 50% 50%;
  background-repeat:   no-repeat;
  background-size:     cover;
}
.title{
  text-align: center;
  padding: 0% 9%;
}

.article{
  width:30%;
  margin: 2% 0%;
  background-color: orange;
}