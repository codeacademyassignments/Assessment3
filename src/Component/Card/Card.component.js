import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.component.css';
import image from '../../Images/cover_image.jpeg';
import CardFooter from './CardFooter/CardFooter.component';
// import Image from './Image/Image.component';
// import TextContent from './TextContent/TextContent.component';

class Card extends Component {
  render() {
    // const { title } = this.props;
    return (
      <article className="article">
        <div className="img" style={{ backgroundImage: `url(${image})` }} />
        <p className="title">{this.props.title}</p>
        <CardFooter rating={this.props.rating} bookId={this.props.bookId} liked={this.props.liked} onHeartClick={(data) => { alert(data); }} />
      </article>
    );
  }
}


export default Card;
