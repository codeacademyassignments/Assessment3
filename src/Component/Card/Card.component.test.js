import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card.component';
import TestData from '../../TestData/TestData';

const blog = TestData['J K Rowling'][0];
describe('Section', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<Card rating={blog.rating} liked title={blog.Name} bookId={blog.id} />);
    expect(tree).toMatchSnapshot();
  });
});
