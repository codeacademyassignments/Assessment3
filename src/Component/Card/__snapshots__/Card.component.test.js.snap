// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshot should match 1`] = `
<article
  className="article"
>
  <div
    className="img"
    style={
      Object {
        "backgroundImage": "url(cover_image.jpeg)",
      }
    }
  />
  <p
    className="title"
  >
    Harry Potter and the Sorcerers Stone (Harry Potter, #1)
  </p>
  <div
    className="cardFooter"
  >
    <div>
      <p
        className="rating"
      >
        4.45
      </p>
    </div>
    <div
      className="heartImage"
    >
      <img
        alt="heart-black"
        className="heart"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "heart-red.svg",
          }
        }
      />
    </div>
  </div>
</article>
`;
